### PR TITLE
source-mysql: datatype fixes

### PR DESCRIPTION
**Pull request by postamar**
## What
This PR tries to get source-mysql's datatypes correct by leveraging the improved `MySqlDatatypeIntegrationTest`.

## How
n/a

## Review guide
n/a

## User Impact
Source-mysql will handle some datatypes differently:
- very large DECIMAL values are now rendered properly without loss of precision
- BIT(n) where n > 1 are now properly handled as byte arrays instead of as integers
- FLOAT(?), FLOAT(?,?), DOUBLE(?), DOUBLE(?,?) might render very slightly differently

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt [hunk 1]
```diff
@@ -162,9 +159,9 @@ class MysqlSourceOperations :
             MysqlType.TEXT,
             MysqlType.MEDIUMTEXT,
             MysqlType.LONGTEXT,
-            MysqlType.JSON,
             MysqlType.ENUM,
             MysqlType.SET -> StringFieldType
+            MysqlType.JSON -> JsonStringFieldType
```
[the remark this answers is not in the document]
Comment (rodireich) [reply airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt]: in old mysql:
```
case TINYTEXT, TEXT, MEDIUMTEXT, LONGTEXT, JSON, ENUM, SET -> putString(json, columnName, resultSet, colIndex);
```

[the remark this answers is not in the document]
Comment (postamar) [reply airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt]: oof, OK

[the remark this answers is not in the document]
Comment (postamar) [reply airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt]: I'm thinking that we should fix this in a different PR

File: airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceDebeziumOperations.kt [hunk 2]
```diff
@@ -72,6 +75,22 @@ class MySqlSourceDebeziumOperations(
...
+                    val textNode: TextNode = data[field.id] as? TextNode ?: continue
+                    val bigDecimal = BigDecimal(textNode.textValue()).stripTrailingZeros()
+                    data.put(field.id, bigDecimal)
+                }
+                LeafAirbyteSchemaType.JSONB -> {
+                    val textNode: TextNode = data[field.id] as? TextNode ?: continue
+                    data.set<JsonNode>(field.id, Jsons.readTree(textNode.textValue()))
+                }
+                else -> continue
+            }
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceDebeziumOperations.kt:92]: this feels like a bad abstraction here, the fact that we (the implementor of DebeziumOperations) have to iterate through the schema types. Just some food for thoughts

**Review by stephane-airbyte — APPROVED** [review]

File: airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceDebeziumOperations.kt (hunk 2 above)
Comment (postamar) [reply airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceDebeziumOperations.kt:92]: Yeah, it's not great. Perhaps `Stream` should have pre-computed groupings of `Field`s per Airbyte type. Or per `FieldType`. Not sure.

**Comment by postamar** [comment]
@rodireich I looked into the debezium-connector-mysql code and all of the weird edge cases in the legacy converter logic apply only to debezium snapshots, as far as I can tell, which we just don't do. When using debezium in regular, non-snapshot mode the Debezium engine stays on the happy path with no weird java types.

**Comment by postamar** [comment]
@rodireich actually there is one edge case which this PR doesn't handle correctly: default values! I'm working on adding test cases for these.

**Comment by postamar** [comment]
I added test cases for default values, which proved very useful.

**Pull request merged**